middleware/faq.rst

```diff
@@ -4,6 +4,16 @@ Frequently asked questions
 
 This page contains answer to common questions about the data access and pipeline middleware, such as the Butler and pipeline tasks.
 
+.. note::
```

Comment:
Please remove this entire note because it's inside-baseball and inappropriate for a user-facing product (the rule-of-thumb is that airing "dirty laundry" that's irrelevant to the user's goals is a bad idea).

Instead, wherever you use

```rst
:program:`butler`
```

in the docs, please instead use

```
:ref:`lsst.daf.butler-scripts <butler>
```

and the equivalent for `pipetask`. I.e., make the docs work for the user rather than trying to use the program role that doesn't serve any purpose. Someday the program role or a successor to it will give us the combination of the cross-reference and the right typography. But for now, making the correct link is more useful.

To make this easier we can add a Sphinx substitution to make this cross-reference. I'll make that change in the Documenteer-based configuration module so that the substitution can be used both here and in package documentation.

Reply:
It may be dirty laundry, but I actually think some of this is very important information for any consumer of the middleware docs; now that I've made it slightly easier to find the `daf_butler` scripts page (other changes on this ticket), it'd be very easy to find that first and then get very confused by the fact that some of the subcommands aren't listed there (or on any one page, for that matter).  How about this:
 - "How do I find the middleware command-line documentation?" becomes its own FAQ (it wouldn't be the only FAQ here describing workarounds for a known defect);
 - I should add notes to at least the `daf_butler` and `obs_base` scripts pages linking to each other, and noting that there may be other subcommands elsewhere.
 - Note in both the FAQ entry and those notes that the best way to find CLI docs is to just use `--help`, because that _is_ a complete list.
 - I drop the references to the Sphinx linking limitation; I agree that's not something the reader is probably not interested (unless they're the kind that tends to file doc-improvement tickets, I suppose).

Reply:
Sounds reasonable to me. Suggest a question rephrase to "Where can I find documentation for command-line butler queries?"